erpnext/hr/doctype/process_payroll/process_payroll.py

```diff
@@ -22,9 +22,9 @@ def get_emp_list(self):
 
 		sal_struct = frappe.db.sql("""
 				select name from `tabSalary Structure`
-				where docstatus != 2 and payment_account = '%(payment_account)s' and 
+				where docstatus != 2 and company = '%(company)s' and 
```

Comment:
Avoid using quote (`'%(company)s'`) in query. Write without quote and pass the arguments with comma (instead of %).

Your query will look like:

```
sal_struct = frappe.db.sql("""
	select name from `tabSalary Structure`
	where docstatus != 2 and company = %(company)s and 
	ifnull(salary_slip_based_on_timesheet,0) = %(salary_slip_based_on_timesheet)s""", 
	{"company": self.company, "salary_slip_based_on_timesheet":self.salary_slip_based_on_timesheet})
```